codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/error/OperationErrorGenerator.kt

```diff
@@ -120,39 +115,46 @@ class OperationErrorGenerator(
                 "StdError" to RuntimeType.StdError,
                 "ErrorMeta" to errorMetadata,
             ) {
-                rust(
-                    """
-                    Self::Unhandled({
-                        let mut builder = #T::builder().source(source);
-                        builder.set_meta(meta);
-                        builder.build()
-                    })
-                    """,
-                    unhandledError(runtimeConfig),
+                rustTemplate(
+                    """Self::Unhandled(#{Unhandled} { source, meta: meta.unwrap_or_default() })""",
+                    "Unhandled" to unhandledError(runtimeConfig),
                 )
             }
         }
-        writer.rustBlock("impl #T for ${errorSymbol.name}", RuntimeType.Display) {
+    }
+
+    private fun RustWriter.renderImplDisplay(errorSymbol: Symbol, errors: List<StructureShape>) {
+        rustBlock("impl #T for ${errorSymbol.name}", RuntimeType.Display) {
             rustBlock("fn fmt(&self, f: &mut ::std::fmt::Formatter<'_>) -> ::std::fmt::Result") {
-                delegateToVariants(errors) {
-                    writable { rust("_inner.fmt(f)") }
+                delegateToVariants(errors) { variantMatch ->
+                    when (variantMatch) {
+                        is VariantMatch.Unhandled -> writable { rust("f.write_str(\"unhandled error\")") }
```

Comment:
do we lose the code in this case? The unhandled error should still include the error code if we have it

Reply:
The previous behavior only outputted "unhandled error", but it's trivial to add the code to it, so I just went ahead and did it.